src/imitation/algorithms/bc.py

```diff
@@ -52,6 +52,94 @@ def __call__(self, _):
         return self.lr
 
 
+class EpochOrBatchIteratorWithProgress:
+    def __init__(
+        self,
+        data_loader: Iterable[dict],
+        n_epochs: Optional[int] = None,
+        n_batches: Optional[int] = None,
+        on_epoch_end: Optional[Callable[[], None]] = None,
+    ):
+        """Wraps DataLoader so that all BC batches can be processed in a one for-loop.
+
+        Also uses `tqdm` to show progress in stdout.
+
+        Args:
+            data_loader: An iterable over data dicts, as used in `BC`.
+            n_epochs: The number of epochs to iterate through in one call to
+                __iter__. Exactly one of `n_epochs` and `n_batches` should be provided.
+            n_batches: The number of batches to iterate through in one call to
+                __iter__. Exactly one of `n_epochs` and `n_batches` should be provided.
+            on_epoch_end: A callback function without parameters to be called at the
+                end of every epoch.
+        """
+        if n_epochs is not None and n_batches is None:
+            self.mode = "epochs"
```

Comment:
Nit: since `self.mode` can only take two values legally, can we make this explicitly -- with an `Enum`, or just making it a boolean flag (e.g. `self.use_epochs`)?

Reply:
Made a boolean flag